recurring_contract/tests/test_recurring_contract.py

```diff
@@ -0,0 +1,189 @@
+#    Releasing children from poverty in Jesus' name
+#    @author: Albert SHENOUDA <albert.shenouda@efrei.net>
+#
+#    The licence is in the file __openerp__.py
+#
+##############################################################################
+
+from openerp.tests import common
+from datetime import datetime
+from openerp.tools import DEFAULT_SERVER_DATE_FORMAT as DF
+from openerp import netsvc
+import logging
+logger = logging.getLogger(__name__)
+
+
+class test_recurring_contract(common.TransactionCase):
+
+    """
+        Test Project recurring contract. It's the first test file.
+        We are testing the first easier scenario :
+            - we are creating one contract
+            - a payment option in which:
+                - 1 invoice is generated every month
+                - with 1 month of invoice generation in advance
+        We are testing if invoices data are coherent with data in the
+        associate contract.
+    """
+
+    def setUp(self):
+        super(test_recurring_contract, self).setUp()
+        # Creation of an account
+        account_type = self.registry('account.account.type').search(
+            self.cr, self.uid, [
+                ('close_method', '=', 'unreconciled'),
+            ])[0]
+        property_account_receivable = self.registry('account.account').search(
+            self.cr, self.uid, [
+                ('type', '=', 'receivable'),
+                ('user_type', '=', account_type)
+            ])[0]
+        property_account_payable = self.registry('account.account').search(
+            self.cr, self.uid, [('type', '=', 'payable')])[0]
+
+        # Creation of a partner
+        partner = self.registry('res.partner')
+        partner_id = partner.create(self.cr, self.uid, {
+            'name': 'Monsieur Client 137',
+            'property_account_receivable': property_account_receivable,
+            'property_account_payable': property_account_payable,
+        })
+        # Creation of payement terms
+        payment_term = self.registry('account.payment.term')
+        payment_term_id = payment_term.search(self.cr, self.uid, [
+            ('name', '=', '15 Days')
+        ])[0]
+        # Creation of payment option then contract and its line
+        self.contract_group0 = self._create_group('do_nothing', 1,
+                                                  'month', partner_id, 1,
+                                                  payment_term_id)
+
+        self.contract_id = self._create_contract(
+            datetime.today().strftime(DF), self.contract_group0,
+            datetime.today().strftime(DF))
+        self.contract_line_id = self._create_contract_line(self.contract_id,
+                                                           '40.0')
+
+    def _create_contract(self, start_date, group_id, next_invoice_date):
+        """
+            Create a contract. For that purpose we have created a partner
+            to get his id
+        """
+        # Creation of an account
```

Comment:
Lines 76-96 are copy-paste from lines 36-54. You can remove them and use the partner_id you created in SetUp().